library/src/com/google/maps/android/kml/KmlContainerParser.java

```diff
@@ -0,0 +1,204 @@
+package com.google.maps.android.kml;
+
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+
+/**
+ * Parses the container of a given KML file into a KmlContainer object
+ */
+/* package */ class KmlContainerParser {
+
+    private final static int START_TAG = XmlPullParser.START_TAG;
+
+    private final static int END_TAG = XmlPullParser.END_TAG;
+
+    private final static String PROPERTY_REGEX = "name|description|visibility|open";
+
+    private final static String CONTAINER_REGEX = "Folder|Document";
+
+    private final static String PLACEMARK = "Placemark";
+
+    private final static String STYLE = "Style";
+
+    private final static String STYLE_MAP = "StyleMap";
+
+    private final static String EXTENDED_DATA = "ExtendedData";
+
+    private final static String GROUND_OVERLAY = "GroundOverlay";
+
+    private final KmlFeatureParser mFeatureParser;
+
+    private final XmlPullParser mParser;
+
+    private KmlContainer mContainer;
+
+
+
+    /* package */ KmlContainerParser(XmlPullParser parser) {
+
+        mParser = parser;
+        mFeatureParser = new KmlFeatureParser(parser);
+        mContainer = null;
+    }
+
+    /**
+     * Creates a new Container object (created if a Document or Folder start tag is read by the
+     * XmlPullParser) and assigns specific elements read from the XmlPullParser to the container.
+     */
+
+    /* package */ void createContainer() throws XmlPullParserException, IOException {
+        KmlContainer kmlContainer = new KmlContainer();
+        assignContainerProperties(kmlContainer);
+        mContainer = kmlContainer;
+    }
+
+    /**
+     * Obtains relevant values from an XML start tag and assigns these values to variables
+     * within a KmlContainer class
+     *
+     * @param kmlContainer Container to store XML start tags and its corresponding value
+     */
+    /* package */ void assignContainerProperties(KmlContainer kmlContainer)
```

Comment:
apart from testing, this should be private.  Is there a way to test without calling this method directly?  Could you just call createContainer instead?